src/main/java/com/google/devtools/build/lib/actions/LocalHostCapacity.java

```diff
@@ -23,27 +23,32 @@
  */
 @ThreadCompatible
 public final class LocalHostCapacity {
+  private static final double EFFECTIVE_CPUS_PER_HYPERTHREADED_CPU = 0.6;
 
   private static final OS currentOS = OS.getCurrent();
   private static ResourceSet localHostCapacity;
 
   private LocalHostCapacity() {}
 
   public static ResourceSet getLocalHostCapacity() {
+    return getLocalHostCapacity(EFFECTIVE_CPUS_PER_HYPERTHREADED_CPU);
+  }
+
+  public static ResourceSet getLocalHostCapacity(double cpusPerHyperthreadedCpuFactor) {
     if (localHostCapacity == null) {
-      localHostCapacity = getNewLocalHostCapacity();
+      localHostCapacity = getNewLocalHostCapacity(cpusPerHyperthreadedCpuFactor);
```

Comment:
This is incorrect - it's caching the first cpusPerHypterthreadedCpuFactor parameter permanently. I also don't think this is the right way to go about it. I'd suggest for LocalHostCapacity to always return the true number of threads / cores, and then post-process that over in the --jobs option handling.

Reply:
Why is there caching in the first place? In a virtualized machine, there is a chance that this changes, right?

Reply:
I'm not sure why this is cached here - maybe for performance? In any case, downstream is not currently written in a way that would allow it to handle changes during a build. (Also, there are physical machines that support CPU hotplugging. ;-)